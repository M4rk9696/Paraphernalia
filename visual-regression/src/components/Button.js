import React from "react";
import styled from "styled-components";

const StyledButton = styled.div`
  width: 100px;
  padding: 5px;
  border: 2px solid;
  border-radius: 10px;
  background-color: #eee;
  text-align: center;
  line-height: 50px;

  &:hover {
    cursor: pointer;
  }
`;

export default ({ name, action }) => {
  return (
    <>
      <StyledButton datatest-id="button" onClick={action}>
        {name}
      </StyledButton>
    </>
  );
};
