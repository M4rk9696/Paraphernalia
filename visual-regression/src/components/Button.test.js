import React from 'react'
import Button from "./Button";
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from "@testing-library/react";

describe("test button", () => {
  // it("should be able to render", () => {
  //   const mockFn = jest.fn();

  //   const {asFragment} = renderer(<Button name="test" action={mockFn} />);

  //   expect(asFragment()).toMatchSnapshot()
  // })

  it("should be able to render", () => {
    const mockFn = jest.fn();

    const tree = renderer.create(<Button name="test" action={mockFn} />).toJSON();

    expect(tree).toMatchSnapshot()
  })

  it("should be able to click a button", async () => {
    const mockFn = jest.fn();

    render(<Button name="test" action={mockFn} />);

    fireEvent.click(screen.getByText("test"));

    expect(mockFn).toHaveBeenCalled();
  });

});
