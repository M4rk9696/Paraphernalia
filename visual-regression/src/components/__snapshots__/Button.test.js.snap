// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test button should be able to render 1`] = `
<div
  className="sc-bdfBwQ hUNnmt"
  onClick={[MockFunction]}
>
  test
</div>
`;
